stock_picking_analytic/models/stock_picking.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2021 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    _name = "stock.picking"
+    _inherit = ["stock.picking", "analytic.mixin"]
+
+    original_analytic_distribution = fields.Json(
+        help="Technical field. Use to store the value of"
+        "analytic_distribution if there is no lines",
+    )
+    analytic_distribution = fields.Json(
+        inverse="_inverse_analytic_distribution",
+        states={
+            "done": [("readonly", True)],
+            "cancel": [("readonly", True)],
+        },
+    )
+    # Assign a dummy search arg to the field to avoid triggering the warning.
+    # Without this, use of move_ids_without_package in @api.depends() of the compute
+    # method would trigger a warning from
+    # https://github.com/odoo/odoo/blob/ae94f14a844352f66490ec326fe2d1a716023891/odoo/fields.py#L808-L812  # noqa
```

Comment:
If you use a shorter hash, line won't be as long and noqa won't be required, e.g.:

```suggestion
    # https://github.com/odoo/odoo/blob/ae94f14a/odoo/fields.py#L808-L812
```